src/main/java/uk/gov/hmcts/reform/profilesync/service/impl/ProfileUpdateServiceImpl.java

```diff
@@ -38,85 +38,75 @@
+    private  ProfileSyncAuditDetails syncUser(String bearerToken, String s2sToken,
+                                              String userId, UserProfile updatedUserProfile,
+                                              ProfileSyncAudit syncAudit)
+            throws UserProfileSyncException {
 
-        log.info(loggingComponentName, "Inside  syncUser:: method");
+        log.info("{}:: Inside  syncUser method ::{}", loggingComponentName);
         Response response = userProfileClient.syncUserStatus(bearerToken, s2sToken, userId, updatedUserProfile);
-
-        log.info(loggingComponentName, "Body response::" + response.body().toString());
+        String message = "success";
         if (response.status() > 300) {
-
-            log.error(loggingComponentName, "Exception occurred while updating the user profile: Status - {}"
-                    + updatedUserProfile.getIdamStatus());
-            saveSyncJobAudit(response.status(), "fail");
-            throw new UserProfileSyncException(HttpStatus.valueOf(response.status()), "Failed to update");
-
+            log.error("{}:: Exception occurred while updating the user profile: Status - {}" + response.status(),
+                    loggingComponentName);
+            message = "the user profile failed while updating the status";
+            log.error("{}:: Body response::{}" + response.body(), loggingComponentName);
+            syncAudit.setSchedulerStatus("fail");
         }
-
-        log.info(loggingComponentName, "Successfully updated the user profile: Status - {}");
-    }
-
-    private void saveSyncJobAudit(Integer idamResponse, String message) {
-
-        SyncJobAudit syncJobAudit = new SyncJobAudit(idamResponse, message, Source.SYNC);
-        syncJobRepository.save(syncJobAudit);
+        return  new ProfileSyncAuditDetails(new ProfileSyncAuditDetailsId(syncAudit,userId),response.status(),message,
+                LocalDateTime.now());
     }
 
-    public Map<Map<String, Boolean>, IdamStatus> idamStatusResolver() {
+    public  String resolveIdamStatus(StringBuilder stringBuilder) {
 
-        Map<Map<String, Boolean>, IdamStatus> idamStatusMap = new HashMap<Map<String, Boolean>, IdamStatus>();
-        idamStatusMap.put(addRule(false, true), IdamStatus.PENDING);
-        idamStatusMap.put(addRule(true, false), IdamStatus.ACTIVE);
-        idamStatusMap.put(addRule(false, false), IdamStatus.SUSPENDED);
-
-        return idamStatusMap;
-    }
-
-    public Map<String, Boolean> addRule(boolean activeFlag, boolean pendingFlag) {
-        Map<String, Boolean> pendingMapWithRules = new HashMap<>();
-        pendingMapWithRules.put("ACTIVE", activeFlag);
-        pendingMapWithRules.put("PENDING", pendingFlag);
-        return pendingMapWithRules;
+        switch (stringBuilder.toString().toLowerCase()) {
+            case "falsetrue":
+                return IdamStatus.PENDING.name();
```

Comment:
not getting why we are using switch case ap<Map<String, Boolean>, IdamStatus> idamStatusMap = new HashMap<Map<String, Boolean>, IdamStatus>();

Map is clean code.

Reply:
where did u find the map code .. i dint see it.

Reply:
it's very hard to readability and understanding so make it simple